pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -0,0 +1,623 @@
+package azure
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/api"
+	utilerrors "k8s.io/kubernetes/pkg/util/errors"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/glog"
+)
+
+// GetLoadBalancer returns whether the specified load balancer exists, and
+// if so, what its status is.
+func (az *AzureCloud) GetLoadBalancer(clusterName string, service *api.Service) (status *api.LoadBalancerStatus, exists bool, err error) {
+	lbName := getLoadBalancerName(clusterName)
+	pipName := getPublicIPName(clusterName, service)
+	serviceName := getServiceName(service)
+	glog.Infof("get(%s): START clusterName=%q lbName=%q", serviceName, clusterName, lbName)
+
+	glog.Infof("get(%s): lb(%s) - retrieving", serviceName, lbName)
+	_, err = az.LoadBalancerClient.Get(az.ResourceGroup, lbName, "")
```

Comment:
oh, now I see you are passing it to the `check` function.  Please use different names for clarity.  (here and elsewhere)